### PR TITLE
Proposed classes re-locations

### PR DESCRIPTION
**Pull request by peter-budo**
- Moved JSON classes to core module
- Extracted `asList()` originally in SkillOverview to `skillsListFrom(SkillOverview so)` method of SoldierOverviewFragment (yes, does not belong there, once we have database persistence handling classes it will be moved to corresponding counter part)
- Moved battle report classes to a package (I rather have it in appropriate package then flying in generic package with endless list of other classes)

### CODE REVIEW

**Comment by karllindmark** [comment]
@peter-budo 

I'd like to counter-propose a solution that's suitable for both of us - we still have fragments/activities/etc separation, but we have packages in those which represent the different parts. I do see why one would want to group everything related together, but the new proposed structure itself makes me have to think twice regarding where to find something. Structure looks less cluttered as well as we don't have x packages on the top level that are specific to a certain area of the app. :)

In other words:

```
- activities
    - foobar
        - FoobarActivity.java
- adapters
    - foobar
        - FoobarActivity.java
- base
   - BaseFragment.java
   - BaseLoadingFragment.java
- fragments
   - soldier
        - SoldierOverviewFragment.java
        - AssignmentsFragment.java
-  jsonmodels
    - soldieroverview
        - Class1.java
        - Class2.java
    - assignments
        - Class3.java
        - Class4.java
    - battlereports
        - Class5.java
        - Class6.java
```

Just a small example of what I'm trying to show. :)

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/WeaponStats.java [hunk 1]
```diff
@@ -37,15 +37,4 @@ public double getAccuracy() {
     public int getHeadshotCount() {
         return headshotCount;
     }
-
-    @Override
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/WeaponStats.java:41]: :+1:

File: core/src/test/java/com/ninetwozero/bf4intel/assignments/AssignmentsTest.java [hunk 2]
```diff
@@ -21,7 +21,7 @@ public void should_parse_Json() throws IOException {
         JsonParser parser = new JsonParser();
         JsonObject dataJson = parser.parse(reader).getAsJsonObject().getAsJsonObject("data");
         reader.close();
-        Assignments assignments = gson.fromJson( dataJson, Assignments.class);
+        Assignments assignments = gson.fromJson(dataJson, Assignments.class);
```
Comment (karllindmark) [inline core/src/test/java/com/ninetwozero/bf4intel/assignments/AssignmentsTest.java:25]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/VehicleStats.java [hunk 3]
```diff
@@ -16,12 +16,4 @@ public int getTimeIn() {
     public int getDestroyedEnemySameVehicle() {
         return destroyedEnemySameVehicle;
     }
-
-    @Override
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/VehicleStats.java:20]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/SoldierOverview.java [hunk 4]
```diff
@@ -59,17 +59,4 @@ public int getMaxScoreCurrentRank() {
     public int getScoreLeftToNextRank() {
         return nextRank.getPointsNeeded() - basicSoldierStats.getScore();
     }
-
-    @Override
-    public String toString() {
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/SoldierOverview.java:64]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/Rank.java [hunk 5]
```diff
@@ -27,13 +27,4 @@ public int getPointsNeeded() {
     public String getImageSlug() {
         return "r" + level;
     }
-
-    @Override
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/Rank.java:31]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/PersonaInfo.java [hunk 6]
```diff
@@ -44,16 +44,4 @@ public int getPlatform() {
     public int getGame() {
         return game;
     }
-
-    @Override
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/PersonaInfo.java:48]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/CompletionProgress.java [hunk 7]
```diff
@@ -30,14 +30,4 @@ public int getMaxValue() {
     public int getPercentage() {
         return percentage;
     }
-
-    @Override
-    public String toString() {
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/CompletionProgress.java:35]: :+1:

File: core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/BaseStatsModel.java [hunk 8]
```diff
@@ -58,18 +58,4 @@ public int getServiceStarProgress() {
     public int getKillCount() {
         return killCount;
     }
-
-    @Override
-    public String toString() {
-        return "BaseStatsModel{" +
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/soldieroverview/BaseStatsModel.java:64]: :+1:

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/fragments/SoldierOverviewFragment.java [hunk 9]
```diff
@@ -252,4 +254,15 @@ private void displayCompletions(final View baseView, final List<CompletionProgre
             contentArea.addView(parent);
         }
     }
+
+    private List<Skill> skillsListFrom(SkillOverview so) {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/fragments/SoldierOverviewFragment.java:254]: Good point! :+1:

**Comment by peter-budo** [comment]
LOL you pushing for old organisation that I got you to use when I started to participate. I'm fine with having packages (adapter, base etc.) for base classes that are extended somewhere else, but these days I do not see reason to keep activities and fragments apart if they for united user experience.

I think that json classes can have "super" package name json in core module to indicate what they for

**Comment by karllindmark** [comment]
Could you type up an "example" structure like I did above?

Also, :+1: for "super" package name in core module, if you mean the following (just wanna check to see that I'm following you):

```
- json
    - battlereports
        - Class1.java
        - Class2.java
    - soldieroverview
        - Class3.java
        - Class4.java
```

:-)

**Comment by peter-budo** [comment]
```
- adapter
    - SimpleAdapter
    - ComplexAdapter
- activity
    - BaseActivity
    - ListActivity
- fragment
   - BaseFragment
   - MagicalFragment
- overview
    - OverviewFragment (extending BaseFragment)
    - OverviewAdapter (extending SimpleAdapter)
- assignments
    - AssignmentsActivity (extending BaseActivity)
    - AssignmentsFragment (extending MagicalFragment)
    - AssignmentAdapter (extending ComplexAdapter)
```

When need it overview and assignments can be moved to "super package" like json something along the line of bf3 where we had soldier, platoon, news, feeds

**Comment by karllindmark** [comment]
```
-base
    - adapter
        - SimpleAdapter.java
        - ComplexAdapter.java
    - app
        - BaseActivity.java
        - FooActivity.java
        - BaseFragment.java
        - MagicalFragment.java
- ui
    - overview
        - OverviewFragment (extending BaseFragment)
        - OverviewAdapter (extending SimpleAdapter)
    - assignments
        - AssignmentsActivity (extending BaseActivity)
        - AssignmentsFragment (extending MagicalFragment)
        - AssignmentAdapter (extending ComplexAdapter)
```

**Comment by karllindmark** [comment]
TODO (?)
- Move jsonmodels to core/json/<section>/*
- Move activities & fragments to BF4Droid/ui/<section>/*

**Comment by peter-budo** [comment]
@karllindmark project structure updated

**Comment by peter-budo** [comment]
@karllindmark utils and resources moved to top level

**Pull request merged**